corehq/apps/api/resources/auth.py

```diff
@@ -149,6 +150,18 @@ def dummy(request, domain, **kwargs):
             return response
 
 
+class RequirePermissionAuthenticationInteral(LoginAndDomainAuthentication):
+    def __init__(self, *args, **kwargs):
+        super(RequirePermissionAuthenticationInteral, self).__init__(*args, **kwargs)
+
+    def is_authenticated(self, request, **kwargs):
+        return self._auth_test(request, wrappers=[
+            self._get_auth_decorator(request),
+            wrap_4xx_errors_for_apis,
+            require_permission('view_locations', login_decorator=self._get_auth_decorator(request)),
```

Comment:
this permission should be passed in to the constructor and not hardcoded (like `RequirePermissionAuthentication` below)

Reply:
Understood. That would be a better way to go.
Thanks

Reply:
Done @snopoke

Reply:
nice, can we not just update LoginAndDomainAuthentication to have this permission argument and then not need this new class?

Reply:
having a separate class makes it clearer what's happening. Also LoginAndDomainAuthentication is already doing other permissions checks so it would make the class weird to do different checks if you passed in some permission

Reply:
Sounds okay.
Just to share the thought behind that,
the way other classes that inherit `LoginAndDomainAuthentication` and overrite this method completely, even when there are overlaps in what they all they check, does sound something that will be hard to manage over time. I could not see a way to avoid that earlier but with this new attribute getting added, i thought it was worth thinking about that once and avoid adding another class.

Reply:
It could be an issue if there was a plethora of these classes. A potential way to restructure would be to extract any common functionality to separate functions or mixins (not suggesting we do that now though).